### PR TITLE
feat: use localQuote if we have the curve

### PR DESCRIPTION
**Pull request by michielbdejong**

### CODE REVIEW

File: test/coreSpec.js [hunk 1]
```diff
@@ -434,17 +436,17 @@ describe('Core', function () {
         })
       })
     })
-    it.skip('returns a quote when there is a direct (but remote) path', function * () {
+    it('returns a quote when there is a direct (but remote) path', function * () {
```
Comment (sharafian) [inline test/coreSpec.js]: Can we add tests for when it expects a curve but doesn't have one?

Comment (michielbdejong) [inline test/coreSpec.js]: That's what https://github.com/interledgerjs/ilp-core/pull/89/commits/39beac6ad1d74483a359798200b1635565e1a762#diff-58f3a08270cc753a5a98228e9dcffb7aR367 tests, right? The route has no curve, so remote quoting is used.

Comment (sharafian) [inline test/coreSpec.js]: gotcha, so is the connector deciding between curve/remote quote automatically?

Comment (michielbdejong) [inline test/coreSpec.js]: yes, https://github.com/interledgerjs/ilp-core/pull/89/files#diff-355162647f6ba28669fbaf055ff11029R127

File: src/lib/core.js [hunk 2]
```diff
@@ -121,15 +121,14 @@ class Core extends EventEmitter {
...
 
-    // If we know a local route to the destinationAddress, use the local route.
-    // If the route's destination is exactly the prefix being targeted, use the local route.
-    // Otherwise, ask a connector closer to the destination.
-    // if (isLocalPath || isDirectPath) return localQuote
-    if (isLocalPath) return localQuote
-    if (isDirectPath) debug('quote is direct, but multi-hop-direct is disabled')
+    if (localQuote.sourceAmount && localQuote.destinationAmount) return localQuote
+
+    // Reaching this point means the best next hop is know, but its liquidity curve is not.
```
Comment (sentientwaffle) [inline src/lib/core.js]: s/know/known

**Comment by sharafian** [comment]
Can you test this branch on an ILP Connector using configured routes (`CONNECTOR_ROUTES`)? I suspect it breaks that functionality

**Comment by michielbdejong** [comment]
ilp-connector no longer uses ilp-core for quoting, it has its own quoter. So this PR is actually useless, probably.

**Comment by michielbdejong** [comment]
The commits from this PR live on in the mj-use-local-quote branch.

**Pull request closed without merging**